File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        l1 = len(word1)
        l2 = len(word2)
        l = max(l1, l2)
        res = []
        for i in range(l):
            if i < l1:
                res.append(word1[i])
            if i < l2:
                res.append(word2[i])
        return "".join(res)
        

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int i = 0;
        for(; i< Math.min(word1.length(), word2.length()); i++) {
            sb.append(word1.charAt(i));
            sb.append(word2.charAt(i));
        }
        String ss = sb.toString();
        if(word1.length() > word2.length()) {
            return ss + word1.substring(i);
        } else if(word1.length() < word2.length()) {
            return ss + word2.substring(i);
        }
        return ss;
    }
}